src/color.js

```diff
@@ -286,21 +286,35 @@ exports.byElement = function() {
   }, null, null);
 };
 
-exports.bySS = function() {
+exports.bySS = function(grad) {
+  var palette;
+  if (grad) {
+    palette = {
+      C: grad._colors[0],
+      H: grad._colors[1],
+      E: grad._colors[2],
+    };
+  } else {
+    palette = {
+      C: [0.8, 0.8, 0.8, 1.0],
+      H: [0.6, 0.6, 0.9, 1.0],
+      E: [0.2, 0.8, 0.2, 1.0]
+    };
+  }
 
   return new ColorOp(function(atom, out, index) {
     switch (atom.residue().ss()) {
       case 'C':
-        out[index] = 0.8;   out[index+1] = 0.8; 
-        out[index+2] = 0.8; out[index+3] = 1.0;
+        out[index] = palette.C[0];   out[index+1] = palette.C[1]; 
```

Comment:
I'll update those later two suggestions. For `color.bySS`, I had originally
written it as a dictionary, but when i plugged it into a deep page with
buttons to switch color method, it made more sense to keep the parameter
consistent with other functions.
On Jul 13, 2016 11:42 AM, "Marco Biasini" <notifications@github.com> wrote:

In src/color.js <https://github.com/biasmv/pv/pull/162#discussion_r70684533>
:

>
>    return new ColorOp(function(atom, out, index) {
>      switch (atom.residue().ss()) {
>        case 'C':
> -        out[index] = 0.8;   out[index+1] = 0.8;
> -        out[index+2] = 0.8; out[index+3] = 1.0;
> +        out[index] = palette.C[0];   out[index+1] = palette.C[1];

The switch statement is no longer required. You can now write this as

var ss = atom.residue().ss();var color = palette[ss];if (color !== undefined) {
  out[index] = color[0];
  out[index+1] = color[1];
  out[index+2] = color[2];
  out[index+3] = color[3];
}

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub
<https://github.com/biasmv/pv/pull/162/files/822ac5ee98d861fff189960450647742722829a1#r70684533>,
or mute the thread
<https://github.com/notifications/unsubscribe/AHenfZDGSnRPFVKXkxvM1h8cPO9noqiqks5qVTGvgaJpZM4JLr35>
.